### DISCUSSION TITLE
Using stream on `non-nullable` lists

### DISCUSSION THREAD

**Discussion by saihaj** [graphql/defer-stream-wg#25] [Q&A] [answered]
```graphql
type NestedObject {
  nullableSlow: [String]
  nonNullableSlow: [String]!
}

type Query {
  nested: NestedObject
}
```
If I try to `@stream` `nullableSlow`  it will work but when trying on `nonNullableSlow ` I get the following error:
```json
{
  "errors": [
    {
      "message": "Stream directive cannot be used on non-list field \"nonNullableSlow\" on type \"NestedObject\".",
      "locations": [
        {
          "line": 3,
          "column": 21
        }
      ]
    }
  ]
}
```

Here is a codesandbox: https://codesandbox.io/s/stream-error-non-null-list-ntgem?file=/src/index.mjs

**Comment by robrichard** [comment, marked as answer]
@saihaj the validation rule `StreamDirectiveOnListFieldRule` was not properly checking non-null wrapping types. This is fixed now on the `defer-stream` branch
